public/grpc-client.js

```diff
@@ -32,23 +32,27 @@ async function getCredentials(lndSettingsDir) {
   return grpc.credentials.createSsl(lndCert);
 }
 
-function getMetadata(lndSettingsDir) {
-  const metadata = new grpc.Metadata();
-  const macaroonPath = path.join(lndSettingsDir, 'admin.macaroon');
-  const macaroonHex = fs.readFileSync(macaroonPath).toString('hex');
-  metadata.add('macaroon', macaroonHex);
-  return metadata;
+function getMacaroonCreds(lndSettingsDir, network) {
+  return grpc.credentials.createFromMetadataGenerator(function(args, callback) {
+    const metadata = new grpc.Metadata();
+    const macaroonPath = path.join(
+      lndSettingsDir,
+      `data/chain/bitcoin/${network}/admin.macaroon`
+    );
+    const macaroonHex = fs.readFileSync(macaroonPath).toString('hex');
+    metadata.add('macaroon', macaroonHex);
+    callback(null, metadata);
+  });
```

Comment:
How does this work with return if it's asynchronous? Don't we need to wait for the callback to set the metadata?

Reply:
In order to combine the ssl credentials with the macaroon metadata, we need to turn the metadata into a `CallCredentials` object. 

`createFromMetadataGenerator` requires an metadata generator (the asynchronous function) and returns a `CallCredentials` object that can be composed with the ssl `ChannelCredentials` object: https://github.com/GoogleCloudPlatform/google-cloud-node/issues/1346#issuecomment-240912476

Reply:
Ok. Thanks for the explanation.